widgets/htdocs/alphafold/index.js

```diff
@@ -0,0 +1,245 @@
+
+  static get properties() {
+    return {
+      loadCompleted: { state: true },
+      exons: { state: true },
+      variants: { state: true },
+      proteinFeatures: { state: true },
+      selectedExonIndices: { state: true },
+      selectedSiftIndices: { state: true },
+      selectedPolyphenIndices: { state: true },
+      selectedProteinFeatureIndices: { state: true },
+    }
+  }
+
+  constructor() {
+    super();
+    this.selectedExonIndices = [];
+    this.selectedSiftIndices = [];
+    this.selectedPolyphenIndices = [];
+    this.selectedProteinFeatureIndices = {};
+  }
+
+  // prevent the component from rendering into the shadow DOM
+  createRenderRoot() {
+    return this;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    Promise.all([
+      this.loadPdbeMolstarScript(),
+      this.fetchData()
+    ]).then(([, result]) => {
+      const { alphafoldId, exons, variants, proteinFeatures } = result;
+      this.initializeMolstar(alphafoldId);
+      this.exons = exons;
+      this.variants = variants;
+      this.proteinFeatures = proteinFeatures;
+      this.selectedProteinFeatureIndices = Object.keys(proteinFeatures)
+        .reduce((obj, key) => {
+          obj[key] = [];
+          return obj;
+        }, {});
+    }).catch(error => {
+      this.onLoadFailed();
+    });
+  }
+
+  updated(updatedProperties) {
+    const relevantPropertyNames = [
+      'selectedExonIndices',
+      'selectedSiftIndices',
+      'selectedPolyphenIndices',
+      'selectedProteinFeatureIndices'
+    ];
+    const updatedPropertyNames = [...updatedProperties.keys()];
+    const isRelevantPropertyUpdated = relevantPropertyNames
+      .some(name => updatedPropertyNames.includes(name));
+
+    if (isRelevantPropertyUpdated && this.loadCompleted) {
+      this.updateMolstarSelections();
+    }
+  }
+
+  loadPdbeMolstarScript() {
+    return new Promise((resolve, reject) => {
+      const script = document.createElement('script');
+      script.type = 'text/javascript';
+      script.src = 'https://alphafold.ebi.ac.uk/assets/js/af-pdbe-molstar-plugin-1.1.1.js';
+      script.onload = resolve;
+      script.onerror = reject;
+      document.head.appendChild(script);
+    });
+  }
+
+  async fetchData() {
+    const { restUrlRoot, enspId } = this.dataset;
+    const alphafoldId = await fetchAlphaFoldId({ rootUrl: restUrlRoot, enspId });
+    const exons = await fetchExons({ rootUrl: restUrlRoot, enspId });
+    const variants = await fetchVariants({ rootUrl: restUrlRoot, enspId });
+    const proteinFeatures = await fetchProteinFeatures({ rootUrl: restUrlRoot, enspId });
+
+    return {
+      alphafoldId,
+      exons,
+      variants,
+      proteinFeatures
+    };
+  }
+
+  initializeMolstar(afdbId) {
+    this.molstarInstance = new PDBeMolstarPlugin();
+
+    const options = {
+      customData: {
+        url: `https://alphafold.ebi.ac.uk/files/${afdbId}-model_v1.cif`,
```

Comment:
Seen this url being used at different places. Should we move it to to global space?

Reply:
like you did using featureTypeToUrlPatternMap

Reply:
Yes, good point! I didn't realise that both the assets and molecule files are hosted on the same site. Updated.